src/ambianic/pipeline/store.py

```diff
@@ -131,6 +131,71 @@ def _save_sample(self,
         self.notify(save_json)
         return image_path, json_path
 
+    def _save_sample_new(self,
+                     inf_time=None,
+                     image=None,
+                     thumbnail=None,
+                     inference_result=None,
+                     inference_meta=None):
+        time_prefix = inf_time.strftime("%Y%m%d-%H%M%S.%f%z-{suffix}.{fext}")
+        image_file = time_prefix.format(suffix='image', fext='jpg')
+        image_path = self._output_directory / image_file
+        thumbnail_file = time_prefix.format(suffix='thumbnail', fext='jpg')
+        thumbnail_path = self._output_directory / thumbnail_file
+        json_file = time_prefix.format(suffix='inference', fext='json')
+        json_path = self._output_directory / json_file
+        inf_json = []
+        if inference_result:
+            for inf in inference_result:
+                label, confidence, box, leaning_angle, keypoint_corr = inf
```

Comment:
@bhavikapanara as discussed on our call, the model specific inference classes (fall_detect.py, object_detect.py, etc.) should provide an inference object that is readily [json serializable](https://stackoverflow.com/questions/3768895/how-to-make-a-class-json-serializable)
.

store.py should just be left with the responsibility to **store** the inference data. It should just dump the provided inference data structure to json with one line of code:

```
inf_json.append(one_inf)
```

store.py should not have to know the details of formatting all possible types of inference outputs. As we add more inference types in the future, store.py will get too bloated and harder to maintain.

For now we can leave the logic in store.py to lookup a thumnail attribute and save it to disk along with the json dump, but in a next iteration I would like you to propose a more generic way to handle that. For example fall detection should save the pair of images that were used to determine a positive detection. We need them saved for debugging and also to present to users , so they are aware what the ground truth is (start of fall, end of fall).